### DISCUSSION TITLE
OpenHab ebus Bridge

### DISCUSSION THREAD

**Discussion by martingwb** [john30/ebusd#741] [Q&A]
Hi,
After receiving the adapter and successfully reading/writing from/on the ebus of by Vaillant geotherm I would like to see if/how an integration into OpenHab is possible.
Before going to the OpenHab community I wanted to check if someone can share about a possible route cause for these messages which appear after creating and connecting to the OpenHab ebus Bridge:

> 2022-12-05 17:20:00.580 [bus error] device status: unexpected available enhanced byte 2
2022-12-05 17:20:00.601 [bus error] device status: unexpected available enhanced following byte 1

The log is showing 15-20 of these lines per second.

Thank you for any clues and hints.

**Comment by mwildbolz** [comment]
Hi martingwb!
Maybe a quick answer to your first question:
There are 2 ways for integrating ebusd with openHAB - I tried both of them:
- Usage of the ebus binding from csowada (see [on Github](https://github.com/csowada/openhab-ebus-binding/releases))
- integration using MQTT protocol

I started using the ebus binding from csowada which worked OK, but recently switched completely to MQTT.
Big advantage in my opinion is, that you can rely on the ebus-configuration in terms of message definition when using MQTT. If you use the openHAB binding, you have to manually add new definitions in the configuration files of the binding, which is (my personal meaning) not as straight forward as the definition for ebusd.
And, I'm using MQTT also for other parts of openHAB, so no additional effort on this side...

Regarding your second topic:
No clue, why you get this errors. Has to be clarified with others in regards to ebusd and the used hardware...

BR,
MW

**Comment by karwak** [comment]
Hi martingwb,

as mwildbolz described, using MQTT works also very well for me for already some years.

However, so far I am using ebusd solely for reading values. Never tried to actively set a value.

BG
karwak
